sklearn/ensemble/_forest.py

```diff
@@ -2633,7 +2634,32 @@ def fit_transform(self, X, y=None, sample_weight=None):
         super().fit(X, y, sample_weight=sample_weight)
 
         self.one_hot_encoder_ = OneHotEncoder(sparse=self.sparse_output)
-        return self.one_hot_encoder_.fit_transform(self.apply(X))
+        output = self.one_hot_encoder_.fit_transform(self.apply(X))
+        self._n_features_out = output.shape[1]
+        return output
+
+    def get_feature_names_out(self, input_features=None):
+        """Get output feature names for transformation.
+
+        Parameters
+        ----------
+        input_features : array-like of str or None, default=None
+            Only used to validate feature names with the names seen in :meth:`fit`.
+
+        Returns
+        -------
+        feature_names_out : ndarray of str objects
+            Transformed feature names, in the format of randomtreesembedding_{i}_{j},
+            where i is the tree used to generate the leaf and j is the leaf index.
+        """
+        check_is_fitted(self, "_n_features_out")
+
+        feature_names = [
+            f"randomtreesembedding_{i+1}_{j}"
```

Comment:
```suggestion
            f"randomtreesembedding_{tree}_{leaf}"
```

I would still go with 0 indexing with trees. It is consistent with all other names. The leaf indices will be all over the place because it depends on where it appears in the tree.